File: week-03/day-1/Workshop 17.4/src/PrintEachLine.java
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.List;

public class PrintEachLine {
    public static void main(String[] args) throws IOException {
        // Write a program that opens a file called "my-file.txt" and prints each
        // line from the file to separate lines on the console.
        // If the program is unable to read the file (for example the file does not exist),
        // it should print the following error message: "Unable to read file: my-file.txt".
        File file = new File("resources/my-file.txt");
        if (file.exists()){
           Path filePath = Paths.get("resources/my-file.txt");
            List<String> lines = new ArrayList<>();
           lines = Files.readAllLines(filePath);
           for (String greeting: lines) {
               System.out.println(greeting);
            }
        } else {
            System.out.println("Unable to read file: " + file);
        }
    }
}

File: week-07/day-2/Workshop 16.5.2023/print_each_line.py
# Write a program that opens a file called "my-file.txt" and prints
# each line from the file.
# If the program is unable to read the file (for example it does not exist),
# it should print the following error message: "Unable to read file: my-file.txt"

def open_file(filename):
    try:
        my_file = open(filename, 'r')
        for line in my_file:
            print(line.rstrip())
    except IOError:
        print("Unable to read file: " + filename)


open_file('file-name.txt')

open_file('i_do_not_exist.txt')